backend/db.py

```diff
@@ -10,55 +10,57 @@
 )
 from sqlalchemy.orm import declarative_base
 
-Base = declarative_base()
-
-# Heavily inspired by https://praciano.com.br/fastapi-and-async-sqlalchemy-20-with-pytest-done-right.html
-# https://medium.com/@tclaitken/setting-up-a-fastapi-app-with-async-sqlalchemy-2-0-pydantic-v2-e6c540be4308
-
-
-class DatabaseSessionManager:
-    def __init__(self, host: str, engine_kwargs: dict[str, Any] = {}):
-        self._engine = create_async_engine(host, **engine_kwargs)
-        self._sessionmaker = async_sessionmaker(autocommit=False, bind=self._engine)
+from databases import Database
+from sqlalchemy import create_engine
+from sqlalchemy.orm import declarative_base, sessionmaker
 
-    async def close(self):
-        if self._engine is None:
-            raise Exception("DatabaseSessionManager is not initialized")
-        await self._engine.dispose()
-
-        self._engine = None
-        self._sessionmaker = None
+Base = declarative_base()
+class DatabaseConnection:
+    """Manages database connection (sqlalchemy & encode databases)"""
 
-    @contextlib.asynccontextmanager
-    async def connect(self) -> AsyncIterator[AsyncConnection]:
-        if self._engine is None:
-            raise Exception("DatabaseSessionManager is not initialized")
+    def __init__(self):
+        self.database = Database(
+            settings.SQLALCHEMY_DATABASE_URI.unicode_string(), min_size=5, max_size=20
+        )
+        # self.database = Database(settings.DTM_DB_URL.unicode_string())
+        self.engine = create_engine(
```

Comment:
Do we need to create two database connections for some reason? One with encode/databases, the other with SQLAlchemy?

This was also done for DroneTM, but I removed it. The only reason I see for the SQLAlchemy engine is if we need to use alembic too, but this shouldn't be part of the endpoint dependencies and should probably be done elsewhere

Reply:
No Sam, Previously our setup was different and I used get_session() in the migrated endpoints. Now we are using databases so once it is implemented in all those migrated endpoints, I will remove it as removing now will throw error so.